pkg/infinispan/configuration/server/configuration.go

```diff
@@ -147,3 +154,17 @@ func funcMap() template.FuncMap {
 func supportedMajorVersion(v uint64) bool {
 	return v >= 14 && v <= 15
 }
+
+func mapVerstionsToSpec(operand version.Operand, spec *Spec) {
```

Comment:
It's probably beyond the scope of this PR, but I'm wondering if we should add the JGroups version to the Operand metadata provided by `INFINISPAN_OPERAND_VERSIONS`? That way we can handle the schema version automatically without requiring the if/else logic and it means we can move to a new JGroups version without having to make changes to the Operator code. It could also provide benefits for rolling upgrades as the Operator could take into account the JGroups version when allowing/denying a rolling upgrade.

Reply:
I agree with you. We need a declarative way to address it. I opened the followup.
Also, I'm adding the link to the follow up in a TODO.
What do you think of it?

Reply:
Thanks. I've added more information to the issue to try to make it clearer what we want to achieve and why those changes are beneficial.